### PR TITLE
Add selector-nested-pattern; closes #962

### LINKED ISSUE #962: selector-nested-pattern
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A new rule to only allow pseudo classes to be nested. When I write SASS 

> Which rule, if any, is this issue related to?

e.g. `number-leading-zero`

> What CSS is needed to reproduce this issue?

**Valid**

```css
.item {
	 &:before {
	 	content: '';
	 }
}
```

**Invalid**

```css
.item {
	.child {
		background: #FFF;
	}
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "only-nest-pseudo-classes": true 
  }
}
```

> Which version of stylelint are you using?

`5.0.1`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

Using the PostCss plugin

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

SCSS syntax

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/32136155-selector-nested-pattern?utm_campaign=plugin&utm_content=tracker%2F9282518&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F9282518&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by davidtheclark**
Ref #962.

### CODE REVIEW

**Comment by davidtheclark** [comment]
The `dedent` template literal function might be helpful for us in other tests.

File: package.json [hunk 1]
```diff
@@ -74,6 +74,7 @@
     "benchmark": "^2.1.0",
     "common-tags": "^1.3.0",
     "coveralls": "^2.11.9",
+    "dedent": "^0.6.0",
```
Comment (jeddy3) [inline package.json]: Do we want to use this separate package or [`stripIndent`](https://github.com/declandewet/common-tags#stripindent) from the already included `common-tags`?

**Comment by jeddy3** [comment]
Besides that one query about the `dedent` dependency, this looks great!

It feels like it's been an age since either you or I tucked into any new rules or options, and now we've adding them left, right and centre :)

**Comment by davidtheclark** [comment]
> It feels like it's been an age since either you or I tucked into any new rules or options, and now we've adding them left, right and centre :)

I had some time today and thought it'd be a good idea to dive into a few of these!

**Pull request merged**